### PR TITLE
[TRRFFDA-306] Make user registration always work based on Email Notifications

### PR DESCRIPTION
**Pull request by alintreznai**

### CODE REVIEW

File: rdrf/rdrf/models/definition/models.py [hunk 1]
```diff
@@ -633,6 +633,12 @@ def proms_system_url(self):
         except KeyError:
             return None
 
+    def registration_allowed(self):
+        registration_enabled = self.has_feature(RegistryFeatures.REGISTRATION)
+        return (
+            registration_enabled and EmailNotification.objects.filter(registry=self, disabled=False).exists()
```
Comment (sztamas) [inline rdrf/rdrf/models/definition/models.py]: Can we be more specific here?
There are multiple types of email notifications, we would like to check that we have user registration notifications set up.

File: rdrf/rdrf/settings.py [hunk 2]
```diff
@@ -585,7 +585,7 @@
 
 # In case you set up customised email templates for the "new-patient" notification, you should
```
Comment (sztamas) [inline rdrf/rdrf/settings.py:586]: Can we update the comment as well?
It should say that this turns off django-registration-redux's email notifications as we would like to send email registration through trrf's email notification system.

**Review by sztamas — CHANGES_REQUESTED** [review]

File: rdrf/rdrf/models/definition/models.py (hunk 1 above)
Comment (alintreznai) [reply rdrf/rdrf/models/definition/models.py]: Updated.

File: rdrf/rdrf/settings.py (hunk 2 above)
Comment (alintreznai) [reply rdrf/rdrf/settings.py:586]: Updated.

**Review by sztamas — APPROVED** [review]

**Pull request merged**